src/Tools/CodeGenerator/Featurizers.yaml

```diff
@@ -1414,58 +1414,50 @@ featurizers:
     num_output_columns: 1
     has_dynamic_output: true
     description: |-
-      Computes lags and leads for values. Works for general time series data sets.
-      For sparse time series data sets, forward fill imputer will be applied so input time series would always be continuous
-      and have a consistent time interval between rows.
+      Copying values from back or in the future per grain. Works for general time series data sets.
 
-      Input type for this featurizer is a tuple of time point of current row, target column to lag or lead
-      and the horizon origin for the row.
+      Offsets are the lag or lead orders we use to find the row in the past or in the future. Positive offset means looking into the future and negative offset means looking back in this featurizer.
+      
+      Horizon is an unsigned int to mark the starting position to look for the data, ranging from 1 to N.
 

```

Comment:
The Horizon represents the maximum value in a range [1, N), where each element in that range is a delta applied to each offset. The resulting matrix will be in the form:

[ 
  [value[1 + offset[0]], value[2 + offset[0]], ..., value[N -1 + offset[0]],
  [value[1 + offset[1]], value[2 + offset[1]], ..., value[N -1 + offset[1]],
  ...
  [value[1 + offset[K - 1]], value[2 + offset[K - 1]], ..., value[N-1 + offset[K - 1]]
]

The resulting matrix size is K rows x N cols, where K is the number of offsets and N is the horizon.

(I think that I have this backwards)